### ISSUE TITLE
Input Dialog wrap EditText in TextInputLayout

### ISSUE THREAD

**Issue by andrewlord1990** [afollestad/material-dialogs#690]
The `TextInputLayout` in Appcompat-v7 support library adds floating hint labels to an `EditText`.

It would mean that in the input dialog the hint would float above the `EditText` when typed in.

It also adds error support to the `EditText` for an error being shown underneath (as in the material design guidelines).

I would be happy to add these to the library, it should be a fairly simple change. Just wanted to check on how you think the API would work.

The floating hint label just requires the `EditText` to be wrapped in a `TextInputLayout`, so that part is fairly simple. I imagine this would work by adding a method to the builder to show floating hint label.

```
inputFloatingHintLabel()
```

Then, for showing errors you need to call `setErrorEnabled(boolean)` on the `TextInputLayout`. Then, to show an error you call `setError(CharSequence)`. This part is slightly more complicated to fix into the builder, as currently the `onInput` callback is only provided with the `MaterialDialog` and the text entered. There would therefore be two approaches for this.

1) A separate method is added to the builder for setting a different input callback.
 ```
inputCallback(new InputErrorCallback() {
    public void onInput(MaterialDialog dialog, TextInputLayout textInput, CharSequence textEntered) {
        ....
    }
})
```
This would then allow the user to call `textInput.setError(errorMessage)`

2) A method is added to `MaterialDialog` to show an input error. This would mean the error can be set from the existing `InputCallback` `onInput` method. By doing:
```
onInput(MaterialDialog dialog, CharSequence text) {
    dialog.setInputErrorMessage(errorMessage);
}
```

I think the second option is less disruptive and fits better into the API. However, wanted to get your thoughts. If you are happy with this feature, I will go ahead and implement it.

Thanks.

**Comment by afollestad** [comment]
We actually chose to not include other libraries for the input dialog (e.g. the Design Support Library or another library that provides similar functionality for EditTexts) on purpose. 

Not everyone would use this and it would increase the size of their app for no reason. For the counter that's displayed when you limit input length, we copied what Google shows on their design guidelines. The same could be done for an error indicator, but this isn't top priority right now.
